src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine.cs

```diff
@@ -66,92 +65,165 @@ public async Task FindReferencesAsync(ISymbol symbol, CancellationToken cancella
             {
                 await using var _ = await _progressTracker.AddSingleItemAsync(cancellationToken).ConfigureAwait(false);
 
```

Comment:
logic for the engine has been effectively rewritten.  previously it was: compute entire inheritance/cascade graph.  use that to determine *entire* set of documetns to search in the solution, then bucket by project and analyze projects as a whole, and individual docs in that project as a whole.

Now it is:

determine initial cascading/inheritance set.  Walk the relevant dependent projects, updating that set as we walk along.  For each project, determine the set of docs to analyze it in and then analyze each across all the finders.